elastic/kibana
# [Discover] Do not update defaultIndex in case of insufficient permissions

## Summary

This PR prevents Discover from updating the `defaultIndex` uiSettings which can lead to a 403 error message when the user doesn't have permissions. `getDefaultDataView` of  `DataViewsService` returns the needed `defaultIndex` also without updating, when the user has insufficient permissions. Works also in Dashboard, Visualizations ...

### Testing
* Add demo date e.g. **Kibana eCommerce orders**
* Create a user having a role that has only readonly permissions to Discover.
* Remove the `defaultIndex` value in Advanced settings. 
* Login with the new user
* When navigating to Discover, there should be no error toast
 

Fixes #46124

### Checklist
